Source/WebCore/testing/MockGamepadProvider.cpp

```diff
@@ -106,8 +121,14 @@ bool MockGamepadProvider::disconnectMockGamepad(unsigned index)
 
     m_connectedGamepadVector[index] = nullptr;
 
-    for (auto& client : m_clients)
-        client->platformGamepadDisconnected(*m_mockGamepadVector[index]);
+    auto gamepadToRemove = m_mockGamepadVector[index].get();
+    for (auto& client : m_clients) {
+        auto gamepadsForClient = m_invisibleGamepadsForClient.find(client);
+        if (gamepadsForClient != m_invisibleGamepadsForClient.end() && gamepadsForClient->value.contains(gamepadToRemove))
```

Comment:
Similarly here, the double look up of gamepadToRemove in contains() and remove() is still unfortunate.

This could we written as:
```
if (gamepadsForClient == m_invisibleGamepadsForClient.end() || !gamepadsForClient->value.remove(gamepadToRemove))
    client->platformGamepadDisconnected(*m_mockGamepadVector[index]);

Reply:
Good point